### PR TITLE
feat: add menu and routing

### PR DESCRIPTION
**Pull request by mritense**
Adds configurable routing functionality, a responsive menu and page layout and accessibility improvements. Closes https://ritense.tpondemand.com/entity/30877-portal-verticaal-menu

### CODE REVIEW

File: packages/user-interface/src/components/menu-button/menu-button.tsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-);
+const MenuButton = () => {
+  const {showMenu} = useContext(LayoutContext);
+
+  return (
+    <Button onClick={showMenu}>
+      <FormattedMessage id="header.menuButton" />
+    </Button>
+  );
+};
```
Comment (bddjong) [inline packages/user-interface/src/components/menu-button/menu-button.tsx:15]: This MenuButton is used in the header on the mobile variant. Clicking it opens the menu (with overview, messages, cases). According to the design however, in order to open the menu, the user must press the blue dropdown menu:
image

**Review by bddjong — CHANGES_REQUESTED** [review]

File: packages/user-interface/src/components/menu-item/menu-item.module.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  &:after {
+    display: none !important;
+  }
+}
+
+.denhaag-menu-button__icon {
+  display: flex;
+  margin-inline-end: var(--nlportal-space-inline-md, 1rem);
+}
```
Comment (bddjong) [inline packages/user-interface/src/components/menu-item/menu-item.module.scss:20]: I think the icon is slightly too large. In the design, the font-size for the menu button label is `20px`, and the icon (container) is also `20px`.

**Comment by bddjong** [comment]
Also a smaller note: I can't scroll down the menu in the scenario where there's a lot of menu buttons. I can imagine it might overflow with multiple opened dropdowns. Not an issue right now with just a few pages however

File: packages/user-interface/src/components/menu-item/menu-item.module.scss (hunk 2 above)
Comment (mritense) [reply packages/user-interface/src/components/menu-item/menu-item.module.scss]: Fixed in latest commit, used 1rem  (in design I saw icons were 15x15, 1rem is close enough I think)

**Comment by mritense** [comment]
> Also a smaller note: I can't scroll down the menu in the scenario where there's a lot of menu buttons. I can imagine it might overflow with multiple opened dropdowns. Not an issue right now with just a few pages however

Fixed in latest commit

File: packages/app/src/styles/nl-portal-design-tokens.css [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
   --nlportal-header-height: 72px;
   --nlportal-header-height-large: 85px;
   --nlportal-header-max-logo-width: 150px;
   --nlportal-header-facet-height: 14px;
+  --nlportal-header-background-color: var(--denhaag-color-white);
+  --nlportal-desktop-menu-width: 230px;
+  --nlportal-menu-item-height: 3rem;
+  --nlportal-menu-background-color: var(--denhaag-color-white);
+  --nlportal-menu-button-active-color: var(--denhaag-menu-button-active-color);
+  --nlportal-menu-icon-size: 1rem;
```
Comment (bddjong) [inline packages/app/src/styles/nl-portal-design-tokens.css:22]: Is it not more accurate to set this to `var(--denhaag-menu-button-font-size)`. That way it's the exact same size as in the design, and it ensures the menu icons scale nicely if another municipality uses a different font size for menu buttons.

File: packages/user-interface/src/components/menu-item/menu-item.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      })}
+    >
+      {item.icon && <div className={styles['denhaag-menu-button__icon']}>{item.icon}</div>}
+      <FormattedMessage id={`pageTitles.${item.titleTranslationKey}`} />
+      {item.showMessagesCount && (
+        <div className={styles['denhaag-menu-button__counter']}>
+          <BadgeCounter>{messagesCount}</BadgeCounter>
+        </div>
+      )}
+    </Link>
```
Comment (bddjong) [inline packages/user-interface/src/components/menu-item/menu-item.tsx:37]: This conditional rendering should be slightly modified to also ensure there are actually messages. In our usage guidelines for the Badge Counter it's noted that the counter should not be visible if the number of messages is equal to 0.

image

**Review by bddjong — CHANGES_REQUESTED** [review]
Two more small comments, sorry I didn't spot the badge counter usage earlier

File: packages/app/src/styles/nl-portal-design-tokens.css (hunk 3 above)
Comment (mritense) [reply packages/app/src/styles/nl-portal-design-tokens.css]: Fixed in latest commit

File: packages/user-interface/src/components/menu-item/menu-item.tsx (hunk 4 above)
Comment (mritense) [reply packages/user-interface/src/components/menu-item/menu-item.tsx]: Fixed in latest commit

**Review by bddjong — APPROVED** [review]

**Pull request merged**